src/MSBuildLocator/Utils/SemanticVersionParser.cs

```diff
@@ -0,0 +1,162 @@
+﻿#if NETCOREAPP
+
+using System;
+using System.Linq;
+
+namespace Microsoft.Build.Locator
+{
+    internal class SemanticVersionParser
```

Comment:
Please attribute this to the version of the NuGet code you looked at (in case we need to follow some changes they make)